components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -250,6 +254,13 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
                 associateTransientRequestData(request, responseWrapper, context);
             }
 
+            // Check if the login access to the application is enabled.
+            if (!checkIfApplicationAccessEnabled(request, responseWrapper, context)) {
+                FrameworkUtils.sendToRetryPage(request, responseWrapper, context,
```

Comment:
Whats the api response we get for app native authentication when this happens?

Reply:
Improved the auth response for app native auth: https://github.com/wso2/carbon-identity-framework/pull/5707/commits/da5a9a273fcaa4f3433cd522ee03c8a2e7618b06